### ISSUE TITLE
Error: Names must be unique in building gcamdata in Rstudio

### ISSUE THREAD

**Issue by sshreekar** [JGCRI/gcamdata#1155]
I am getting the following error when building the gcamdata in Rstudio

[1] "module_emissions_L212.unmgd_nonco2"
Error: Names must be unique.
Run `rlang::last_error()` to see where the error occurred.

**Comment by sshreekar** [comment]
Tracing the error:

 rlang::last_trace()
<error/vctrs_error_names_must_be_unique>
Names must be unique.
Backtrace:
     █
  1. └─gcamdata::driver(write_outputs = FALSE, write_xml = TRUE)
  2.   └─gcamdata:::run_chunk(chunk, all_data[input_names])
  3.     ├─base::do.call(chunk, list(driver.MAKE, all_data))
  4.     └─gcamdata:::module_emissions_L212.unmgd_nonco2(...)
  5.       └─`%>%`(...)
  6.         ├─base::withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
  7.         └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  8.           └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
  9.             └─gcamdata:::`_fseq`(`_lhs`)
 10.               └─magrittr::freduce(value, `_function_list`)
 11.                 └─function_list[[i]](value)
 12.                   └─gcamdata:::rename_SO2(., A_regions)
 13.                     └─`%>%`(...)
 14.                       ├─base::withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
 15.                       └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
 16.                         └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
 17.                           └─gcamdata:::`_fseq`(`_lhs`)
 18.                             └─magrittr::freduce(value, `_function_list`)
 19.                               └─function_list[[i]](value)
 20.                                 ├─dplyr::rename(., Non.CO2 = SO2_name)
 21.                                 └─dplyr:::rename.data.frame(., Non.CO2 = SO2_name)
 22.                                   └─tidyselect::vars_rename(names(.data), !!!enquos(...))
 23.                                     └─tidyselect:::rename_impl(NULL, .vars, quo(c(...)), strict = .strict)
 24.                                       ├─tidyselect:::with_subscript_errors(...)
 25.                                       │ ├─base::tryCatch(...)
 26.                                       │ │ └─base:::tryCatchList(expr, classes, parentenv, handlers)
 27.                                       │ │   └─base:::tryCatchOne(expr, names, parentenv, handlers[[1L]])
 28.                                       │ │     └─base:::doTryCatch(return(expr), name, parentenv, handler)
 29.                                       │ └─tidyselect:::instrument_base_errors(expr)
 30.                                       │   └─base::withCallingHandlers(...)
 31.                                       └─vctrs::vec_as_names(names, repair = "check_unique")
 32.                                         └─vctrs:::validate_unique(names)
 33.                                           └─vctrs:::stop_names_must_be_unique(which(duplicated(names)))
 34.                                             └─vctrs:::stop_names(...)
 35.                                               └─vctrs:::stop_vctrs(...)

**Comment by bpbond** [comment]
@sshreekar Is this on `master`? What if any changes have been made? What's your `sessionInfo()`? Basically, we need some details on your setup.

**Comment by pralitp** [comment]
This is related to a change in behavior in the latest `tidyverse` package on which gcamdata relies.   We will merge the fix soon.  In the mean time here are the fixes to solve the problem (and is backwards compatible with older versions of the `tidyverse` packages as well):
```
diff --git a/input/gcamdata/R/module-helpers.R b/input/gcamdata/R/module-helpers.R
index 0d9edf229..735c99a49 100644
--- a/input/gcamdata/R/module-helpers.R
+++ b/input/gcamdata/R/module-helpers.R
@@ -82,7 +82,8 @@ rename_SO2 <- function(x, so2_map, is_awb = FALSE) {
     # pull so2_map information into SO2 data
     select(region, SO2_name) %>%
     left_join_error_no_match(data_so2, ., by = "region") %>%
-    rename(Non.CO2 = SO2_name) %>%
+    mutate(Non.CO2 = SO2_name) %>%
+    select(-SO2_name) %>%
     bind_rows(data_notso2)
 }
 
diff --git a/input/gcamdata/R/zchunk_L101.nonghg_en_USA_S_T_Y.R b/input/gcamdata/R/zchunk_L101.nonghg_en_USA_S_T_Y.R
index 3d32974dd..bf01edfe1 100644
--- a/input/gcamdata/R/zchunk_L101.nonghg_en_USA_S_T_Y.R
+++ b/input/gcamdata/R/zchunk_L101.nonghg_en_USA_S_T_Y.R
@@ -80,7 +80,7 @@ module_emissions_L101.nonghg_en_USA_S_T_Y <- function(command, ...) {
               spread(L1321.in_EJ_R_cement_F_Y, year, value),
               spread(L124.in_EJ_R_heat_F_Yh, year, value),
               spread(L111.Prod_EJ_R_F_Yh, year, value),
-              spread(L144.in_EJ_R_bld_serv_F_Yh, year, value) %>% rename(sector = service)) %>%
+              spread(L144.in_EJ_R_bld_serv_F_Yh, year, value) %>% select(-sector) %>% rename(sector = service)) %>%
       mutate(technology = fuel) ->
       temp
 
diff --git a/input/gcamdata/R/zchunk_LA100.IEA_downscale_ctry.R b/input/gcamdata/R/zchunk_LA100.IEA_downscale_ctry.R
index 3e3127626..fccee1c47 100644
--- a/input/gcamdata/R/zchunk_LA100.IEA_downscale_ctry.R
+++ b/input/gcamdata/R/zchunk_LA100.IEA_downscale_ctry.R
@@ -142,7 +142,7 @@ module_energy_LA100.IEA_downscale_ctry <- function(command, ...) {
       # the USSR and separated countries thereafter, resulting in unrealistic fuel shares.
       USSR_YUG_SPLIT_YEAR <- 1990
       USSR_YUG_YEARS <- HISTORICAL_YEARS[HISTORICAL_YEARS < USSR_YUG_SPLIT_YEAR]
-      POST_USSR_YUG_YEARS_IEA <- HISTORICAL_YEARS[HISTORICAL_YEARS >= USSR_YUG_SPLIT_YEAR]
+      POST_USSR_YUG_YEARS_IEA <- as.character(HISTORICAL_YEARS[HISTORICAL_YEARS >= USSR_YUG_SPLIT_YEAR])
       L100.USSR_Yug <- filter(L100.IEAcomposite, COUNTRY %in% c("Former Soviet Union (if no detail)",
                                                                 "Former Yugoslavia (if no detail)"))
 
@@ -207,7 +207,7 @@ module_energy_LA100.IEA_downscale_ctry <- function(command, ...) {
       POST_USSR_YUG_YEARS_IEA <- intersect(POST_USSR_YUG_YEARS_IEA, names(L100.USSR_Yug_ctry))
       L100.USSR_Yug_ctry %>%
         select(iso, FLOW, PRODUCT, IEAcomposite, POST_USSR_YUG_YEARS_IEA) %>%
-        left_join_keep_first_only(select(L100.USSR_Yug, one_of("COUNTRY", "FLOW", "PRODUCT", USSR_YUG_YEARS)),
+        left_join_keep_first_only(select(L100.USSR_Yug, one_of("COUNTRY", "FLOW", "PRODUCT", as.character(USSR_YUG_YEARS))),
                                   by = c("IEAcomposite" = "COUNTRY", "FLOW", "PRODUCT")) %>%
         left_join_keep_first_only(select(L100.USSR_Yug_ctry_FLOW_PRODUCT_1990, iso, PRODUCT, `1990_share`),
                                   by = c("iso", "PRODUCT")) ->
```

**Comment by sshreekar** [comment]
I used gcamdata from its git-repo. It is on master. I suspect it is version conlict. I recently update to R version 3.6.2 (2019-12-12) -- "Dark and Stormy Night". Other than that I did not make any changes.

**Comment by pralitp** [comment]
To be clear, it is the updated R -- or really updated R packages that caused the failure rather than the version of gcamdata you are using.

**Comment by sshreekar** [comment]
Well, I updated R and after that I had installed tidyverse package. Then I tried to build gcamdata from locally saved folder. It gave the error. Then, I downloaded the gcamdata from the git and tried it again. It gave the same error.

**Comment by qingyihou** [comment]
@sshreekar, did you solve the error?  I have the same error,

**Comment by pralitp** [comment]
This fix has now been included in #1156.  If you update or install fresh version of `gcamdata` you should now be able to run without these errors.
